numba/unicode.py

```diff
@@ -1388,6 +1389,42 @@ def impl(s):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L11975-L12006    # noqa: E501
+@overload_method(types.UnicodeType, 'isalnum')
+def unicode_isalnum(data):
+    """Implements UnicodeType.isalnum()"""
+
+    def impl(data):
+        length = len(data)
+
+        if length == 1:
+            code_point = _get_code_point(data, 0)
+            if data._is_ascii:
+                return _Py_ISALNUM(code_point)
+            result = (_PyUnicode_IsNumeric(code_point) or
```

Comment:
temporary variable isn't needed? just return the result of evaluating the condition?

Reply:
In the case of ascii, you can do without a temporary variable, because the function _Py_ISALNUM has already implemented a check for both alpha and numeric

Reply:
you can just do this: https://github.com/stuartarchibald/numba/commit/17f58014789c4159f2a6c131799e0c86e5721fa7

Reply:
Done